connectors/db/vector.py

```diff
@@ -189,52 +260,99 @@ def init_vector_store(self):
         self.search_providers = [
             MMRSearchProvider(self.store),
             SimilaritySearchProvider(self.store),
+            HybridSearchProvider(self.store),
         ]
 
     def combine_small_chunks(self, chunks):
         """
-        Combine small chunks into the next chunk
-
-        Sometimes we see the text splitter create a chunk containing only a single line (like
-        a header), we will store these small chunks on the next chunk to avoid storing a
-        document with small context
+        Keep merging small chunks into the next one until the result is large enough.
         """
-        for idx, chunk in enumerate(chunks):
+        merged_chunks = []
+        buffer = ""
+
+        for chunk in chunks:
+            if len(chunk.strip()) == 0:  # Ignore empty chunks
+                continue
+
             if len(chunk) < 200:
-                # this chunk is less than 200 chars, move it to the next chunk
-                try:
-                    chunks[idx + 1] = f"{chunk}\n\n{chunks[idx + 1]}"
-                except IndexError:
-                    # we've reached the end and there is no 'next chunk', just give up
-                    break
-                # make note of its index and pop it later...
-                chunks[idx] = "<<removed>>"
+                buffer += chunk + "\n\n"  # Collect small chunks
+            else:
+                if buffer:
+                    chunk = buffer + chunk  # Merge collected buffer into this chunk
```

Comment:
We need to make sure we do not exceed a max here right? What if we have 40 chunks less than 200 chars in a row. We could in theory end up with one large 8000 char chunk (maybe highly unlikely at the moment..?)

Reply:
Good call. I added code to enforce max size.